update.sh

```diff
@@ -1,39 +1,60 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+set -e # Exit on error
+set -u # Error if an unassigned variable is called
 
 # Detect the operating system
 os=$(uname)
+APP_BASE_NAME=RimSort
+SUPPORTED_OSES=("Darwin" "Linux")
+
+pretty_print_array_values() {
+    local IFS=", "
+    echo "$*"
+}
+
+for supported_os in "${SUPPORTED_OSES[@]}"; do
+    if [[ "$supported_os" == "$os" ]]; then
+        os_is_supported=true
+    fi
+done
+
+if [[ ! $os_is_supported ]]; then
+    echo "The operating system in use \"${os}\" is unsupported, please use once of the following ($(pretty_print_array_values "${SUPPORTED_OSES[@]}"))"
```

Comment:
Currently looks like that:
```
The operating system in use Linux is unsupported, please use once of the following (Darwin,Linux)
```
Let me know if you would like me to replace `Darwin` with `macOS`, I thought about doing that but it'll force me to use a solution which won't look as tidy as the one that is implemented

Reply:
Speaking as one of two resident Mac guys, the name "Darwin" doesn't mean anything to most people. It should be macOS. But that's just my take. I'd like to know what @Ant-V thinks as well.

As for the other part, can we get away with just saying "Linux"? I'm _not_ a resident Linux guy, but aren't there several Linux-based operating systems out there? Just a question.

Reply:
> Speaking as one of two resident Mac guys, the name "Darwin" doesn't mean anything to most people. It should be macOS. But that's just my take. I'd like to know what @Ant-V thinks as well.
> 
> As for the other part, can we get away with just saying "Linux"? I'm _not_ a resident Linux guy, but aren't there several Linux-based operating systems out there? Just a question.

Regarding macOS, I'll let you guys decide
Regarding Linux, there are a ton of different distros, but currently we don't distinguish between them, and support them all.